concepts/eventing-code-guidelines.md

```diff
@@ -389,6 +389,64 @@ MarkReady(apiRuleNew) // instead consider using another function that explicitly
+- Each log message should have enough context to convey what happened.
+- Each log message should have the proper log level ([reference](https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/unified-approach-to-logging-levels.md)).
+
+Code-specific rules:
+
+- Consider using `uber-go/zap` logging library. Label the logger by naming it with the component name and add the context whenever it is possible ([reference](https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure)):
+  <details>
+        <summary>Example</summary>
+
+  ```go
+  namedLogger := r.logger.WithContext().Named("logger-name").With("backend", "BEB")
+  r.namedLogger().Info("Creating Event Publisher deployment")
+  ```
+  will output:
+  ```
+  {"level":"INFO","timestamp":"2022-07-04T12:00:36+02:00","logger":"logger-name","caller":"backend/reconciler.go:741","message":"Creating Event Publisher deployment","context":{"backend":"BEB"}}
+  ```
+  </details>
+
+
+- Don't log the error in case you return the same error as a result for the `Reconcile()` method. **Reason:** kubebuilder will output it too, so in the end the user will have two very similar logs one after another:
+  <details>
+      <summary>Example</summary>
+
+    ```go
+    namedLogger.Errorw("Failed to sync BEB subscription", "error", err)
+    updateErr := r.updateSubscription(ctx, subscription, log); updateErr != nil {
+          return ctrl.Result{}, errors.Wrap(err, updateErr.Error())
+    }
+    ```
+  will result into duplication of logs:
+    ```
+    {"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-subscription-reconciler","caller":"beb/reconciler.go:275","message":"Failed to sync BEB subscription","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","error":"prefix not found"}}
+    {"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"9994dd3e-0104-4170-82aa-79df9ec41af1","error":"prefix not found"}}
+    ```
+  </details>
+
+
+- Capitalize the component names, i.e. Event Publisher, or EventingBackend:
+  ```go
+  namedLogger.Debug("Event Publisher deployment not ready...")
+  ```
+- In order to recognize the error logs easily, they should have the following structure:
+  *past tense starting with **Failed to...**, followed by the error wrapped with some meaningful context*:
+    ```
+    namedLogger.Errorw("Failed to update Event Publisher secret", "error", err)
+    ```
+-  Capitalize the first letter of the first word in the logs, including the error logs:
+  ```go
+  namedLogger.Debug("Creating secret for BEB publisher")
+  ```
```

Comment:
nit: The indention is wrong: it needs to have the same indention as the text (which means one more space):
<img width="872" alt="Screenshot 2022-07-06 at 16 38 53" src="https://user-images.githubusercontent.com/8013145/177576643-77e556ce-45a4-4dbf-92b4-c143ee08db71.png">
```suggestion
   ```go
   namedLogger.Debug("Creating secret for BEB publisher")
   ```
```